src/settings/mod.rs

```diff
@@ -131,8 +133,8 @@ impl Settings {
 
         match config.try_into::<Settings>() {
             Ok(settings) => {
-                // TODO: replace this with proper logging when we have it
-                println!("config: {:?}", settings);
+                let logger = logging::init_logging(&settings).unwrap();
```

Comment:
Better than an `unwrap()` is an `expect(...)` that prints some useful error info.

But even better than that, maybe, is propagating the error to the caller, seeing as this is inside a function that already returns errors. Let's try and minimise the number of places that can panic to as few as possible. And maybe one day we'll want to do more thorough unit-testing of the error paths, which can't happen if the code panics on the inside.